src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -399,61 +351,115 @@ public boolean setLocation(Location<World> location, boolean forced) {
         for (net.minecraft.entity.Entity passenger : passengers) {
             passenger.startRiding(thisEntity, true);
         }
-
-        return true;
     }
 
-    @Override
-    public void setLocationAndRotation(Location<World> location, Vector3d rotation, EnumSet<RelativePositions> relativePositions) {
-        setLocationAndRotation(location, rotation, relativePositions, false);
-    }
+    // for sponge internal use only
+    @SuppressWarnings("unchecked")
+    private boolean teleportTo(Location<World> location) {
+        final MinecraftServer server = (MinecraftServer) Sponge.getServer();
+        final net.minecraft.entity.Entity this$ = (net.minecraft.entity.Entity) (Object) this;
```

Comment:
Also the preferred uage would be to inline the casts anyway instead of assigning to an intermediate local variable, Mixin searches for the redundant casts and strips them anyway, so `(Entity)(Object)this` just becomes `this` in the generated bytecode.